### PR TITLE
Fix system index thread pool tests

### LINKED ISSUE #106957: [CI] KibanaThreadPoolTests testUserThreadPoolsAreBlocked failing
Looks like a newly added test that has failed several times.

**Build scan:**
https://gradle-enterprise.elastic.co/s/g5ekegdwqgzic/tests/:modules:kibana:test/org.elasticsearch.kibana.KibanaThreadPoolTests/testUserThreadPoolsAreBlocked

**Reproduction line:**
```
./gradlew ':modules:kibana:test' --tests "org.elasticsearch.kibana.KibanaThreadPoolTests.testUserThreadPoolsAreBlocked" -Dtests.seed=F7C062CDD0353C3 -Dtests.jvm.argline="-Des.concurrent_search=true" -Dtests.locale=es-CR -Dtests.timezone=Europe/Brussels -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.kibana.KibanaThreadPoolTests#testUserThreadPoolsAreBlocked`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testUserThreadPoolsAreBlocked'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.kibana.KibanaThreadPoolTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
org.elasticsearch.common.util.concurrent.EsRejectedExecutionException: rejected execution of TimedRunnable{original=java.util.concurrent.FutureTask@530d4d8b[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@60cf2ab4[Wrapped task = org.elasticsearch.indices.SystemIndexThreadPoolTests$$Lambda/0x00007fb8b3b070d8@6b709bf]], creationTimeNanos=588437891261, startTimeNanos=0, finishTimeNanos=-1, failedOrRejected=false} on TaskExecutionTimeTrackingEsThreadPoolExecutor[name = node_s2/search, queue capacity = 1000, task execution EWMA = 45.9micros, total task execution time = 13.1ms, org.elasticsearch.common.util.concurrent.TaskExecutionTimeTrackingEsThreadPoolExecutor@6d493698[Running, pool size = 5, active threads = 5, queued tasks = 1000, completed tasks = 88]]

  at __randomizedtesting.SeedInfo.seed([F7C062CDD0353C3:A741B5397CCA7CA0]:0)
  at org.elasticsearch.common.util.concurrent.EsRejectedExecutionHandler.newRejectedException(EsRejectedExecutionHandler.java:51)
  at org.elasticsearch.common.util.concurrent.EsAbortPolicy.rejectedExecution(EsAbortPolicy.java:33)
  at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:841)
  at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1376)
  at org.elasticsearch.common.util.concurrent.EsThreadPoolExecutor.execute(EsThreadPoolExecutor.java:72)
  at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:123)
  at org.elasticsearch.indices.SystemIndexThreadPoolTests.fillThreadPoolQueues(SystemIndexThreadPoolTests.java:103)
  at org.elasticsearch.indices.SystemIndexThreadPoolTests.assertThreadPoolsBlocked(SystemIndexThreadPoolTests.java:79)
  at org.elasticsearch.indices.SystemIndexThreadPoolTests.runWithBlockedThreadPools(SystemIndexThreadPoolTests.java:64)
  at org.elasticsearch.indices.SystemIndexThreadPoolTests.testUserThreadPoolsAreBlocked(SystemIndexThreadPoolTests.java:73)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1570)

```

### PR DESCRIPTION
**Pull request by williamrandolph**
One thing causing trouble here was that that, when the test filled the thread pool queue with no-ops, the test would fail if any of those operations were rejected. I'm not sure why that was happening, but since all sorts of stuff might be happening in the background of an integration test, I think we can introduce some flexibility there.

Once I made the change there, I found that the tests against the blocked thread pools were occasionally timing out on the search request, rather than quickly returning an exception. This also leaked threads. The culprit turned out to be that the RandomizedClient might set `maxConcurrentShardRequests` to 1, which made the search request hang every time. I changed the test to use the default value of 5 explicitly.

Finally, when I changed this test from a unit test to an integration test, I neglected to move it to an internalClusterTest directory. This PR fixes that as well.

Fixes https://github.com/elastic/elasticsearch/issues/106957

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/indices/SystemIndexThreadPoolTestCase.java [hunk 1]
```diff
@@ -88,9 +89,14 @@ private void assertThreadPoolsBlocked() {
         assertThat(e2.getMessage(), startsWith("rejected execution of ActionRunnable"));
         var e3 = expectThrows(
             SearchPhaseExecutionException.class,
-            () -> client().prepareSearch(USER_INDEX).setQuery(QueryBuilders.matchAllQuery()).get()
+            () -> client().prepareSearch(USER_INDEX)
+                .setQuery(QueryBuilders.matchAllQuery())
+                // Don't let the test framework set maxConcurrentShardRequests randomly, because the
```
Comment (thecoop) [inline test/framework/src/main/java/org/elasticsearch/indices/SystemIndexThreadPoolTestCase.java:94]: Could we set it to a random value other than 1?

File: modules/kibana/src/internalClusterTest/java/org/elasticsearch/kibana/KibanaThreadPoolIT.java [hunk 2]
```diff
@@ -32,11 +33,11 @@ protected Collection<Class<? extends Plugin>> nodePlugins() {
     public void testKibanaThreadPool() {
         runWithBlockedThreadPools(() -> {
             // index documents
-            String idToDelete = client().prepareIndex(".kibana").setSource(Map.of("foo", "delete me!")).get().getId();
-            String idToUpdate = client().prepareIndex(".kibana").setSource(Map.of("foo", "update me!")).get().getId();
+            String idToDelete = ESIntegTestCase.client().prepareIndex(".kibana").setSource(Map.of("foo", "delete me!")).get().getId();
```
Comment (thecoop) [inline modules/kibana/src/internalClusterTest/java/org/elasticsearch/kibana/KibanaThreadPoolIT.java:36]: This extra reference to `ESIntegTestCase` is a side-effect of the move refactor?

Comment (williamrandolph) [reply modules/kibana/src/internalClusterTest/java/org/elasticsearch/kibana/KibanaThreadPoolIT.java:36]: Good catch -- it appeared in a refactor but isn't needed. I'll remove it.

File: test/framework/src/main/java/org/elasticsearch/indices/SystemIndexThreadPoolTestCase.java (hunk 1 above)
Comment (williamrandolph) [reply test/framework/src/main/java/org/elasticsearch/indices/SystemIndexThreadPoolTestCase.java]: Is `usually() ? SearchRequest.DEFAULT_MAX_CONCURRENT_SHARD_REQUESTS : randomIntBetween(2, 10)` too elaborate?

Comment (thecoop) [reply test/framework/src/main/java/org/elasticsearch/indices/SystemIndexThreadPoolTestCase.java]: That looks ideal to me

**Review by thecoop — APPROVED** [review]

**Pull request merged**